Client/Frontend/Browser/Onboarding/Welcome/WelcomeViewCallout.swift

```diff
@@ -0,0 +1,478 @@
+                $0.setTitle(secondaryButtonTitle, for: .normal)
+                    $0.titleLabel?.font = .preferredFont(for: .title3, weight: .bold)
+                $0.addAction(UIAction(identifier: .init(rawValue: "secondary.action"), handler: { _ in
+                    secondaryAction()
+                }), for: .touchUpInside)
+                $0.alpha = 1.0
+                $0.isHidden = false
+            }
+            
+            contentView.setCustomSpacing(8.0, after: titleLabel)
+            contentView.setCustomSpacing(24.0, after: detailsLabel)
+            contentView.setCustomSpacing(10.0, after: primaryButton)
+        case .ready(let title, let details, let moreDetails):
+            titleLabel.do {
+                $0.text = title
+                $0.textAlignment = .left
+                $0.font = .preferredFont(for: .title3, weight: .bold)
+                $0.alpha = 1.0
+                $0.isHidden = false
+            }
+            
+            detailsLabel.do {
+                $0.text = "\(details)\n\(moreDetails)"
+                $0.font = .preferredFont(forTextStyle: .body)
+                $0.alpha = 1.0
+                $0.isHidden = false
+            }
+            
+            primaryButton.do {
+                $0.alpha = 0.0
+                $0.isHidden = true
+            }
+            
+            secondaryButton.do {
+                $0.alpha = 0.0
+                $0.isHidden = true
+            }
+            
+            contentView.setCustomSpacing(8.0, after: titleLabel)
+            contentView.setCustomSpacing(0.0, after: detailsLabel)
+            contentView.setCustomSpacing(0.0, after: primaryButton)
+        }
+    }
+}
+
+private class CalloutArrowView: UIView {
+    private let maskLayer = CAShapeLayer()
+    
+    override init(frame: CGRect) {
+        super.init(frame: frame)
+    }
+    
+    required init?(coder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+    
+    override func layoutSubviews() {
+        super.layoutSubviews()
+
+        maskLayer.frame = bounds
+        maskLayer.path = createTrianglePath(rect: bounds).cgPath
+        layer.mask = maskLayer
+    }
+    
+    private func createTrianglePath(rect: CGRect) -> UIBezierPath {
+        let path = UIBezierPath()
+        
+        // Middle Top
+        path.move(to: CGPoint(x: rect.midX, y: rect.minY))
+        
+        // Bottom Left
+        path.addLine(to: CGPoint(x: rect.minX, y: rect.maxY))
+        
+        // Bottom Right
+        path.addLine(to: CGPoint(x: rect.maxX, y: rect.maxY))
+        
+        // Middle Top
+        path.addLine(to: CGPoint(x: rect.midX, y: rect.minY))
+        return path
+    }
+}
+
+private class RoundedBackgroundView: UIView {
+    private let cornerRadius: CGFloat
+    
+    init(cornerRadius: CGFloat) {
+        self.cornerRadius = cornerRadius
+        super.init(frame: .zero)
+        
+        layer.cornerRadius = cornerRadius
```

Comment:
set cornerCurve to continuous

Reply:
Fixed at https://github.com/brave/brave-ios/pull/4508/commits/54b09f3605fdf4a55832abeec019bfb4389e297f